.travis.yml

```
@@ -64,7 +72,16 @@ install:
 - travis_retry pip install pytest-xdist
 
 script:
-- py.test -v --cov nipype --cov-config .coveragerc --cov-report xml:cov.xml -c nipype/pytest.ini --doctest-modules nipype -n auto
+- |
+    if [ "$CHECK_TYPE" = "test" ]; then
+        py.test -v --cov nipype --cov-config .coveragerc --cov-report xml:cov.xml -c nipype/pytest.ini --doctest-modules nipype -n auto
+    fi
+- |
+    if [ "$CHECK_TYPE" = "specs" ]; then
+        make check-before-commit
```

```suggestion
        make specs
```

Perhaps we can get rid of the tests, the trailing-spaces and the docs build?

---

Do we not want to fail if any of these changes a file?

---

Yes of course, but why we wouldn't fail with the two following lines? Isn't there where you check which auto files were changed?

---

Sure, but this will also fail if people have trailing whitespace or drop test files in the repo instead of a temporary directory. We do build the docs in Circle, so that's nbd.

Anyway, I'm fine with reducing it to specs, if that's all we really care about.

---

I was just checking whether this was intended, I have no particular opinion as to whether we should check anything else besides the specs. My only concern is Travis-CI build time - if we are able to catch more potential issues, then let's go ahead with `check-before-commit`.

---

Sorry, I just saw you already had committed my suggestion, please feel free to roll back if this will catch more potential issues.